contracts/lib/LibPrices.sol

```diff
@@ -94,45 +104,28 @@ library Prices {
         PriceInstant[24] memory tracerPrices,
         PriceInstant[24] memory oraclePrices
     ) public pure returns (TWAP memory) {
-        uint256 instantDerivative = 0;
+        uint256 totalTimeWeight = 0;
         uint256 cumulativeDerivative = 0;
-        uint256 instantUnderlying = 0;
         uint256 cumulativeUnderlying = 0;
 
         for (uint256 i = 0; i < 8; i++) {
             uint256 currTimeWeight = 8 - i;
             // if hour < i loop back towards 0 from 23.
             // otherwise move from hour towards 0
-            uint256 j = hour < i ? 23 - i + hour : hour - i;
+            uint256 j = hour < i ? 24 - i + hour : hour - i;
 
             uint256 currDerivativePrice = averagePrice(tracerPrices[j]);
             uint256 currUnderlyingPrice = averagePrice(oraclePrices[j]);
 
-            if (currDerivativePrice > 0) {
-                instantDerivative += currTimeWeight;
-                cumulativeDerivative += currTimeWeight * currDerivativePrice;
-            }
-
-            if (currUnderlyingPrice > 0) {
-                instantUnderlying += currTimeWeight;
-                cumulativeUnderlying += currTimeWeight * currUnderlyingPrice;
-            }
-
-            if (instantDerivative == 0) {
-                return TWAP(0, 0);
-            } else {
-                return
-                    TWAP(
-                        PRBMathUD60x18.div(
-                            cumulativeUnderlying,
-                            instantUnderlying
-                        ),
-                        PRBMathUD60x18.div(
-                            cumulativeDerivative,
-                            instantDerivative
-                        )
-                    );
-            }
+            totalTimeWeight += currTimeWeight;
+            cumulativeDerivative += currTimeWeight * currDerivativePrice;
+            cumulativeUnderlying += currTimeWeight * currUnderlyingPrice;
         }
+
+        return
+            TWAP(
+                cumulativeUnderlying / totalTimeWeight,
+                cumulativeDerivative / totalTimeWeight
+            );
```

Comment:
Should this use `PRBMath.div`?

Reply:
Nah I changed this since it actually shouldn't have been. Using WAD maths would mean that it interprets totalTimeWeight as a WAD value and would ensure the result is a WAD, however we want it to actually decrease decimal places if it needs to (which is the opposite of WAD).